xcode/Subconscious/Shared/Components/Notebook/Notebook.swift

```diff
@@ -679,190 +683,131 @@ struct NotebookModel: ModelProtocol {
-                guard detail.address == child else {
-                    return detail
-                }
-                switch detail {
-                case .editor(var description):
-                    description.address = parent
-                    return .editor(description)
-                case .viewer(var description):
-                    description.address = parent
-                    return .viewer(description)
-                case .profile(let description):
-                    return .profile(description)
-                }
-            })
+    static func succeedSaveEntry(
+        state: NotebookModel,
+        environment: AppEnvironment,
+        address: Slashlink,
+        modified: Date
+    ) -> Update<NotebookModel> {
+        // Just refresh note after save, for now.
+        // This reorders list by modified.
+        return update(
+            state: state,
+            actions: [
+                .refreshLists,
+                .detailStack(.succeedSaveEntry(address: address, modified: modified))
+            ],
+            environment: environment
+        )
+    }
 
-            return update(
-                state: state,
-                actions: [
-                    .setDetails(details),
-                    .refreshLists,
-                    .detailStack(.succeedMergeMemo(parent: parent, child: child))
-                ],
-                environment: environment
-            )
-        }
-        /// Retitle success lifecycle handler.
-        /// Updates UI in response.
-        static func succeedUpdateAudience(
-            state: NotebookModel,
-            environment: AppEnvironment,
-            receipt: MoveReceipt
-        ) -> Update<NotebookModel> {
-            /// Find all instances of this model in the stack and update them
-            let details = state.details.map({ (detail: MemoDetailDescription) in
-                guard let address = detail.address else {
-                    return detail
-                }
-                guard address.slug == receipt.to.slug else {
-                    return detail
-                }
-                switch detail {
-                case .editor(var description):
-                    description.address = receipt.to
-                    return .editor(description)
-                case .viewer(var description):
-                    description.address = receipt.to
-                    return .viewer(description)
-                case .profile(let description):
-                    return .profile(description)
-                }
-            })
+    static func succeedMoveEntry(
+        state: NotebookModel,
+        environment: AppEnvironment,
+        from: Slashlink,
+        to: Slashlink
+    ) -> Update<NotebookModel> {
+        return update(
+            state: state,
+            actions: [
+                .detailStack(.succeedMoveEntry(from: from, to: to)),
```

Comment:
If I understand correctly, we removed the update/setDetails(details) logic from this method because it is already handled by `.detailStack(.succeedMoveEntry(from: from, to: to))`.

Is that correct?